### PR TITLE
Introduce base-nnp

### PR DESCRIPTION
**Pull request by dra27**
OCaml 5.00.0 makes `--disable-naked-pointers` mode the default. #20420 prevents the `ocaml-option-nnp` package from being specified for > 5.00.0. In some ways (as was done in the ocaml-multicore project's repo), the `ocaml-option-nnp` should be forced, but this isn't consistent with how the other `ocaml-option-` packages are used, or the base packages.

However, in changing this, packages which didn't support no-naked-pointers mode, and so which _conflicted_ `ocaml-option-nnp` are now installable on OCaml 5.00.0 (e.g. `llvm`) even though they presumably don't work.

This PR introduces `base-nnp.base` which is installed automatically by `ocaml-variants.5.00.0+trunk` (along with `base-unix.base`, etc.) and which is also pulled in if `ocaml-option-nnp` is selected for 4.12-4.14. Packages which conflicted `ocaml-option-nnp` are updated to conflict `base-nnp` instead.

### CODE REVIEW

File: packages/base-nnp/base-nnp.base/opam [hunk 1]
```diff
@@ -0,0 +1,10 @@
+opam-version: "2.0"
+maintainer: "https://github.com/ocaml/opam-repository/issues"
+synopsis: "Naked pointers prohibited in the OCaml heap"
+description: """
+Installed when the compiler does not permit naked pointers in
+the heap. Prior to OCaml 5.00.0, this mode was expressly selected
+by configuring with --disable-naked-pointers. The shared memory
+parallelism added in OCaml 5.00.0 requires this mode.
+"""
+depends: "ocaml" {>= "5.00"} | "ocaml-option-nnp" | "ocaml-variants" {= "4.06.1+no-naked-pointers+flambda"}
```
Comment (kit-ty-kate) [inline packages/base-nnp/base-nnp.base/opam:10]:
```suggestion
depends: "base-domains" | "ocaml-option-nnp" | "ocaml-variants" {= "4.06.1+no-naked-pointers+flambda"}
```

Comment (kit-ty-kate) [reply packages/base-nnp/base-nnp.base/opam:10]: What about adding `ocaml-option-nnpchecker` as well. I know it’s not really "requiring" no-naked-pointers but if you want to constrain against one, then the other should also probably come with it.

Comment (dra27) [reply packages/base-nnp/base-nnp.base/opam:10]: I don't understand what you're suggesting here?

Comment (kit-ty-kate) [reply packages/base-nnp/base-nnp.base/opam]: Something like that:
```suggestion
depends: "base-domains" | "ocaml-option-nnp" | "ocaml-option-nnpchecker" | "ocaml-variants" {= "4.06.1+no-naked-pointers+flambda"}
```

Comment (dra27) [reply packages/base-nnp/base-nnp.base/opam]: That's a contradiction - `ocaml-option-nnpchecker` and `ocaml-option-nnp` can't be installed at the same time? The NNP checker only works if you're _in_ no-naked-pointers mode.

Comment (kit-ty-kate) [reply packages/base-nnp/base-nnp.base/opam]: yes, I know. Internally it’s not in no-naked-pointers mode, but for users it’s the same thing given if it doesn’t work with one, it won’t work with the other either.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**